### ISSUE TITLE
Use github for package repositories

### ISSUE THREAD

**Issue by fmontesi** [jolie/jpm-old#1]
Right now a JPM server has to store the packages themselves. How about we maintain only the index of packages instead, and leave the burden of storing the packages on external servers? I'd love to use github for that like in packagecontrol for Sublime.

Share your thoughts.

**Comment by SimonLarsen** [comment]
I think this is a good idea. Maintaining a central repository is a lot of work and being able to host package yourself makes it a lot easier for developers as well.

I think allowing any public git repository, not just github, is important. From what I understand [git tagging](http://git-scm.com/book/en/v2/Git-Basics-Tagging) should be ideal for tagging specific commits as versions in a host-agnostic way.

**Comment by fmontesi** [comment]
Yes, any git repo would be fine, it should make zero technical difference. And yes, we should look for tags of the form, e.g, v1.1.2.

**Comment by fmontesi** [comment]
Now that I think about it, the package server functionality is also very useful: some may prefer not to use git after all. :-) We should have it all pluggable.
